MatrixKit/Controllers/MXKRoomViewController.m

```diff
@@ -504,9 +504,9 @@ - (void)destroy
     _bubblesTableView.delegate = nil;
     _bubblesTableView = nil;
 
-    // Non live room data sources are not managed by the MXKDataSourceManager
+    // Non live and peeking room data sources are not managed by the MXKDataSourceManager
     // Destroy them once they are no more displayed
-    if (!roomDataSource.isLive)
+    if (!roomDataSource.isLive || roomDataSource.isPeeking)
```

Comment:
Please add a property to handle the ownership of the roomDataSource at MXKRoomViewController level. 
MXKRoomViewController instance should release only the roomDataSource of which its own.